From dae8349425fd1004dc63988e344eeb3ad8b22aaf Mon Sep 17 00:00:00 2001
From: Wyon Bi <bivvy.bi@rock-chips.com>
Date: Wed, 19 Feb 2020 14:15:49 +0800
Subject: [PATCH] drm/rockchip: lvds: avoid PLL unlock error

We need to enable lvds mode before calling phy_power_on.

Change-Id: I38625c44998bde81bb4c98b70b8be5995d64b477
Signed-off-by: Wyon Bi <bivvy.bi@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_lvds.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_lvds.c b/drivers/gpu/drm/rockchip/rockchip_lvds.c
index e95b955b7a12..666bdca1465a 100644
--- a/drivers/gpu/drm/rockchip/rockchip_lvds.c
+++ b/drivers/gpu/drm/rockchip/rockchip_lvds.c
@@ -201,6 +201,9 @@ static void rockchip_lvds_encoder_enable(struct drm_encoder *encoder)
 	if (lvds->panel)
 		drm_panel_prepare(lvds->panel);
 
+	if (lvds->funcs->enable)
+		lvds->funcs->enable(lvds);
+
 	ret = phy_set_mode(lvds->phy, PHY_MODE_VIDEO_LVDS);
 	if (ret) {
 		DRM_DEV_ERROR(lvds->dev, "failed to set phy mode: %d\n", ret);
@@ -209,9 +212,6 @@ static void rockchip_lvds_encoder_enable(struct drm_encoder *encoder)
 
 	phy_power_on(lvds->phy);
 
-	if (lvds->funcs->enable)
-		lvds->funcs->enable(lvds);
-
 	if (lvds->panel)
 		drm_panel_enable(lvds->panel);
 }
-- 
2.35.3

